### PR TITLE
introduce local cluster

### PR DESCRIPTION
**Pull request by phoracek**
This should be merged only after #59

### CODE REVIEW

File: cluster/clean.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash -e
+
+echo 'Cleaning up ...'
+
+./cluster/kubectl.sh delete --ignore-not-found -f deploy/
```
Comment (qinqon) [inline cluster/clean.sh]: no file specified

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,42 @@
+IMAGE_REPO ?= quay.io/nmstate
```
Comment (qinqon) [inline Makefile]: Missing IMAGE_REGISTRY so we can push to local registry

```make
IMAGE_REGISTRY ?= quay.io
IMAGE_REPO ?= $(IMAGE_REGISTRY)/nmstate
```

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+IMAGE_TAG ?= latest
+MANAGER_IMAGE ?= kubernetes-nmstate-manager
+
```
Comment (qinqon) [inline Makefile]: We can compose here the MANAGER_IMAGE_URL to reuse it 
```make
MANAGER_IMAGE_URL ?= $(IMAGE_REPO)/$(MANAGER_IMAGE):$(MANAGER_IMAGE_TAG)
```

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,42 @@
+IMAGE_REPO ?= quay.io/nmstate
+IMAGE_TAG ?= latest
```
Comment (qinqon) [inline Makefile]: We have be using different tag for different images

```make
MANAGER_IMAGE_TAG ?= latest
```

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+check: check-gofmt check-vet
+
+check-gofmt:
+	./hack/check-gofmt.sh
+
+check-vet:
+	./hack/check-vet.sh
+
+build-manager:
+	operator-sdk build $(IMAGE_REPO)/$(MANAGER_IMAGE):$(IMAGE_TAG)
```
Comment (qinqon) [inline Makefile]:
```make
operator-sdk build $(MANAGER_IMAGE_URL)
```

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+push-manager:
+	docker push $(IMAGE_REPO)/$(MANAGER_IMAGE):$(IMAGE_TAG)
```
Comment (qinqon) [inline Makefile]:
```make
docker push $(MANAGER_IMAGE_URL)
```

File: cluster/clean.sh (hunk 1 above)
Comment (phoracek) [reply cluster/clean.sh]: This should deploy everything under the directory.

File: Makefile (hunk 2 above)
Comment (phoracek) [reply Makefile]: Done

File: Makefile (hunk 3 above)
Comment (phoracek) [reply Makefile]: it's not an URL really (misses the protocol prefix), called it MANAGER_IMAGE_FULL_NAME

File: Makefile (hunk 4 above)
Comment (phoracek) [reply Makefile]: Done

File: Makefile (hunk 5 above)
Comment (phoracek) [reply Makefile]: Done

**Comment by phoracek** [comment]
@qinqon works for me. We might want to change the makefile so it has one target to deploy operator on the cluster and one to run it locally. Both dependent on the sync (of RBAC and friends).

**Pull request merged**